lib/models/undertaking.dart

```diff
@@ -88,7 +88,7 @@ class Undertaking {
 
   /// returns a mapping with kv pairs corresponding to Firebase's
   /// possibly a better getter
-  Map<String, dynamic> toMap({keepClasses = false, includeID = true}) {
+  Map<String, dynamic> toMap({keepClasses = false, includeID = false}) {
```

Comment:
event fromMap used to have id as a separate string. Id is now in the Event class, so this is no longer needed

Reply:
sounds like a design flaw if the event class replaces something from its superclass only to do the exact same thing, and i think that breaks the liskov principle. is there a way to abstract it out so tasks and events have the same ID functionality?

Reply:
It does do more than the superclass, forgor to include it in the highlight.
![image](https://github.com/Scrumbags115/plannertarium/assets/130195350/dcfa1940-2bae-4425-b758-d035dd155c34)

Rn tasks and events should handle id in the same way, but it looked like there was older code that had a separate event ID string in some db functions that was screwing with it a bit. This fix should remove the option to have a separate Id string and after this fix tasks and events should handle it the same.

Idk about liskov substitution principle, it seems really hard to not violate. Like, if we have a subclass, it can't add any new method, it cant change existing method's behaviors, and cant have more side effects. I think Liskov might be for something else, cuz there's no way to make an abstraction

Reply:
alright, I think ideally it makes sense for the undertaking class to be "usable" which tasks and events being more specific versions of it, but that will probably be too much effort this late. But this addresses my concern, since I just wanted to make sure the superclass's core functionality wouldn't be changed too much by subclasses.

Reply:
Actually i just realized that the liskov principle still might apply in this case as it is about sub classes and not super classes